osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.security.providers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.spec.AlgorithmParameterSpec;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+
+public class HsmEncryptionProvider extends AbstractEncryptionProvider implements EncryptionProvider {
+
+    private static final String ALGORITHM = "AES/CBC/NoPadding";
+    private static final String PROVIDER = "nCipherKM";
+    private static final String TYPE = "ncipher.sworld";
+    private static final byte[] IV = new byte[16];
+
+    private KeyStore keyStore;
+
+    public HsmEncryptionProvider(File keyStoreFile) {
+        try {
+            super.setKeyFile(keyStoreFile);
+            this.keyStore = KeyStore.getInstance(TYPE, PROVIDER);
+            FileInputStream fIn = new FileInputStream(keyStoreFile);
+            this.keyStore.load(fIn, null);
+        } catch (CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)